packages/css-engine/src/core/to-value.test.ts

```diff
@@ -132,6 +132,33 @@ describe("Convert WS CSS Values to native CSS strings", () => {
     expect(value).toBe("10px 20px 30px 40px");
   });
 
+  test("function", () => {
+    const translate3D = toValue({
+      type: "function",
+      name: "translate3d",
+      seperator: ",",
```

Comment:
do you think we should have separator as an option in the type? is it really like this in CSS? My understanding was that when you see space separated arguments in a function e.g. `drop-shadow(16px 16px 20px red);` `16px 16px 20px red` are not 4 arguments, its one css value just like you have one value in `box-shadow: 60px -16px teal`

Reply:
Its even using a different type: Dimension vs Number and uses Operator. I think this is that way because you are looking at the AST, I would rather learn from CSS spec way of thinking

Reply:
Generally I don't expect css-tree package to be the most acknowledgeable source, css spec should be one

Reply:
The idea was is that args is not an array but a StyleValue type and can be either tuple or layers and rendered based on this info.

Reply:
was just reading on mdn drop-shadow filter:

> Applies the parameter <shadow> as a drop shadow, following the contours of the image. The shadow syntax is similar to <box-shadow> (defined in the [CSS backgrounds and borders module](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_backgrounds_and_borders)), with the exception that the inset keyword and spread parameter are not allowed.

"shadow syntax is similar to <box-shadow>"

That means it is not literally reusing the same implementation, its a standalone one and so they have differences ...

Reply:
I would still look at this more from data perspective than from syntax, meaning the comma vs space thing is a "syntax" concern, though we are not building an AST, we are serializing the data.

Reply:
In terms of data I still think there is a function and there is arguments, in one case its multiple arguments and in another case its just one and this feels to me like one box shadow value

Reply:
> The idea was is that args is not an array but a StyleValue type and can be either tuple or layers and rendered based on this info.

@TrySound so there is always ONE "argument" - that makes sense to me too

Reply:
Sounds good to me, I am going to push an update now 👍🏽

Reply:
For reference, here is something

```js
{
      type: "function",
      name: "drop-shadow",
      args: {
        type: "tuple",
        value: [
          { type: "unit", value: 10, unit: "px" },
          { type: "unit", value: 10, unit: "px" },
          { type: "unit", value: 10, unit: "px" },
          { type: "keyword", value: "red" },
        ],
      },
 }
```

Reply:
args vs arg? if its always just one

Reply:
we can keep it args though and just use one, this is probably more natural for a funciton

Reply:
Args is used in that css parser api